### DISCUSSION TITLE
Found NaNs in A matrix, possibly caused by NaNs ...

### DISCUSSION THREAD

**Discussion by Fioti38** [PMEAL/OpenPNM#2473] [Q&A] [answered]
Hello and thank you for helping us get the best of openPNM,
Any hints on how to solve this problem ? I tried regenerate_model but it didn't work.
![image](https://user-images.githubusercontent.com/104492138/170871764-d036b926-d72c-4bfd-bbc9-7b683093ce09.png)

Thank you so much.

**Comment by jgostick** [comment]
The error message means that some of the conductance values are "nan", like 0/0 or log(-1) or something like that.  It's a bit tricky to know the exact cause without looking at your specific script.  You can try to hunt it down by printing each object.  The right column tells you how many valid values you have, and if you see something like 443/1000, it means there are 547 nans in that array on that object.  For instance, one common case is that your pores are too big, and overlap each other, and the computation of throat length fails.  The solution would be to use large lattice spacing, smaller pores, or a different model for conductance, like cones and cylinders which is tolerate to overlap.  Anyway that is just an example of how this can be caused.

**Reply by Fioti38** [reply]
Thank you for this quick and precise answer, as you explained it very well changing geometry model solves this issue.
Is changing geometry model from spheresandcylinders to conesandcylinders makes a huge difference in conductance values ?

**Comment by jgostick** [comment, marked as answer]
Cones and cylinders is often a better choice actually, but the differences are minor.  The main benefit is that cones and cylnders do not have the geometrical contraints of non-overlapping.  As usual, the best proof is just to ensure your model matches some known values for your materials.

**Reply by Fioti38** [reply]
Thank you Professor,
One last question, if I get this error on an extracted network from XCT images, how to turn the imported geometry to a conesandCylinders model ?
